truss-chains/truss_chains/code_gen.py

```diff
@@ -1,21 +1,40 @@
-import ast
-import json
+"""
+New plan:
+
+* Collapse truss dir in generated code.
+* Move main file completely unchanged (not even needed to remove main).
+* Put it into truss-packages, keep its name, assert there are no name conflicts.
```

Comment:
sorry, just to be clear, this is the `packages` directory in the truss, right? what are the circumstances under which there would be name conflicts? (In chains there isn't really a notion of a user-defined packages directory, right)

Reply:
Good catch: 1) these were working notes and should not be committed 2) I actually didn't check for conflicts.

Chains currently assumes that the everything from the directory in which the entrypoint is defined (i.e. sibling files and nested dirs) could be imported/used (we could call this "workspace"). e.g.:

```
workspace/
  entrypoint.py
  helper.py
  some_package/
    utils.py
    sub_package/
      ...
```

This stuff gets copied into truss's `/packages` and can be imported on the remote as well. Using code *outside* of the workspace is not supported (you can make it a pip-requirement):

```
shared_lib/
  common.py
workspace/
  entrypoint.py
  ...
```

`shared_lib` can only be imported on the remote if its installed as a pip requirement (site-package), it will not be copied from the local host.

Regarding conflicts: is not an issue anymore since truss has the separation of model and packages (though it would be nice to change that actually).